api/lib/domain/models/UserWithActivity.js

```diff
@@ -0,0 +1,9 @@
+class UserWithActivity {
+  constructor({ user, hasAssessmentParticipations, codeForLastProfileToShare }) {
+    this.hasAssessmentParticipations = hasAssessmentParticipations;
+    this.codeForLastProfileToShare = codeForLastProfileToShare;
+    Object.assign(this, user);
```

Comment:
*suggestion** : `this.user = { ...user };`

Reply:
Ici j'utilise le Object.assign pour éviter de réécrire l'ensemble des propriétés du model User. Mon idée est de faire une sorte de composition avec le model User.

`this.user = { ...user };` ne produit pas le même résultat que `Object.assign(this, user);`. 

`this.user = { ...user };` donne : 
```js
{
  user : {
    firstName: 'John'
  }
};
```

`Object.assign(this, user);` donne : 
```js
{
  firstName: 'John'
};
```

Reply:
justement, avec le spread operator c'est pas le cas :
```
> const user = { a: 1 };
> const toto = { ...user };
> toto
{a: 1}
```

Reply:
Dans ce code `this.user = { ...user };` tu stockes le résultat de l'objet spreadé dans une propriété `user`.

Reply:
Ok du coup c'était pas clair pour moi, ce qu'on veut c'est avoir toutes les propriétés du user. à la racine. Je trouve pas ça hyper clair, si on a mieux je prends.

Reply:
c'est quoi l'utilité de faire l'assign dans l'objet racine ?

Reply:
En soi, ça ne me choque pas. Mais ça peut ne pas être évident pour tout le monde je me dis. 

Si l'objet user évolue avec des infos que tu ne veux pas. L'object assign les embarquera tout de même. 

La liste exhaustive de ce qui t'intéresse me semble plus clair. 

Après si il est là pour remplacer le model user qu'on envoie au front et que tout sera nécessaire. Pourquoi pas

Reply:
Vu avec @xav-car @ajoanny et @KaramHaitham. On part sur le Object.assign en attendant de faire évoluer le model User pour en avoir 1 par application.